app/controllers/posts_controller.rb

```diff
@@ -21,8 +21,10 @@ def create
     @post.likes_counter = 0
 
     if @post.save
+      flash[:notice] = 'Post saved!'
       redirect_to user_post_path(cu, @post)
     else
+      flash[:error] = @post.error.messages
       render :new
     end
   end
```

Comment:
- [ ] You're expected to solve the N+1 problem when fetching all posts and their comments for a user. Hence, 

```ruby
@posts = @user.posts #should have been
@posts = @user.posts.includes(:comments)